admin/authContext.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect, createContext, ReactNode } from "react";
+import { signIn, signOut, addAuthObserver } from "../utils/firebase";
+
+type AuthContextType = {
+  user: string;
+  auth: boolean;
+  setUser: React.Dispatch<React.SetStateAction<string>>;
+  setAuth: React.Dispatch<React.SetStateAction<boolean>>;
+  handleSignOut: () => void;
+  handleSignIn: (email: string, password: string) => Promise<void>;
+};
+
+export const AuthContext = createContext<AuthContextType>(
+  (null as unknown) as AuthContextType
+);
+
+export const AuthProvider: React.FunctionComponent<ReactNode> = props => {
+  const [auth, setAuth] = useState(false);
+  const [user, setUser] = useState("");
+
+  const handleSignOut = async () => {
+    const isSignedOut = await signOut();
+    setAuth(!isSignedOut);
+  };
+
+  const handleSignIn = async (email: string, password: string) => {
+    const user = await signIn(email, password);
+    user ? setAuth(true) : setAuth(false);
```

Comment:
en annen måte å gjøre det samme på er `setAuth(!!user)` - `!!` betyr "omform denne verdien fra truthy/falsy til true/false". Det er bare ! to ganger på rad - så ikke _så_ spesielt som man skulle tro.

Reply:
Also - det er kanskje ikke vits å ha `auth` som en egen tilstandsvariabel - du kan alltid utlede det fra om du har et user objekt eller ikke.